app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -1561,7 +1561,9 @@ protected function _saveProducts()
                 }
                 $rowScope = $this->getRowScope($rowData);
 
-                $rowData[self::URL_KEY] = $this->getUrlKey($rowData);
+                if ($urlKey = $this->getUrlKey($rowData)) {
```

Comment:
I would not recommend here to change return type of the method (line 2725, 2739). As user of getUrlKey method I expecting to receive back some url. Can you please change this implementation to check if URL is empty then do not overwrite it.